source/scripting/Checkpoint.cpp

```diff
@@ -0,0 +1,88 @@
+#include "Checkpoint.hpp"
+#include "Native.hpp"
+
+namespace GTA
+{
+    using namespace Math;
+    using namespace System;
+
+    Checkpoint::Checkpoint(CheckpointIcon icon, Math::Vector3 position, Math::Vector3 pointTo, float radius, System::Drawing::Color color)
+        : Checkpoint(icon, position, pointTo, radius, color, 0)
+    {
+    }
+    Checkpoint::Checkpoint(CheckpointIcon icon, Math::Vector3 position, Math::Vector3 pointTo, float radius, System::Drawing::Color color, int reserved) : _icon(icon), _position(position), _pointTo(pointTo), _radius(radius), _color(color), _reserved(reserved)
+    {
+        _handle = Native::Function::Call<int>(Native::Hash::CREATE_CHECKPOINT, static_cast<int>(icon), position.X, position.Y, position.Z, pointTo.X, pointTo.Y, pointTo.Z, radius, static_cast<int>(color.R), static_cast<int>(color.G), static_cast<int>(color.B), static_cast<int>(color.A), reserved);
```

Comment:
I know this is picky, but there is no need to cast the Color.R/G/B/A to an int, casting byte to int is done implicitly